cypress/integration/test.js

```diff
@@ -101,69 +101,19 @@ describe("Testing Example App", function() {
 });
 
 /* gets position of current day on calendar */
-function todaysPosition() {
-  var currentMonth = new Date(Date.now());
-  var today = currentMonth.getDate();
-
-  currentMonth.setDate(1);
-  var firstOfMonth = currentMonth.getDay();
-  var todaysIndex;
-
-  if (firstOfMonth > 0) {
-    todaysIndex = today + (firstOfMonth - 1);
-  } else {
-    // Sunday gives result of 0, but is in position 7 on our calendar
-    todaysIndex = today + (7 - 1);
-  }
-
-  return [Math.ceil(todaysIndex / 7), todaysIndex % 7];
-}
-
 function getToday() {
-  var today = todaysPosition();
-  return cy.get(`tbody > :nth-child(${today[0]}) > :nth-child(${today[1]})`);
-}
-
-/* Determines position of next day based on current day
-  and whether it's the last day of the week or month */
-function nextDayPosition(currentDay) {
-  return [
-    lastDay() ? 1 : currentDay[0],
-    currentDay[1] === 7 ? 1 : currentDay[1] + 1
-  ];
-}
+  const currentMonth = new Date();
+  const today = currentMonth.getDate();
+  const day = currentMonth.getDay();
 
-function getNextDay(currentDay) {
-  var nextDay = nextDayPosition(currentDay);
+  currentMonth.setDate(1);
+  const firstOfMonth = currentMonth.getDay();
+  const todaysIndex = today + (firstOfMonth === 0 ? 7 : firstOfMonth) - 1;
 
-  if (lastDay()) {
-    return cy
-      .get("#next-month")
-      .click()
-      .get(`tbody > :nth-child(${nextDay[0]}) > :nth-child(${nextDay[1]})`);
-  } else {
-    return cy.get(
-      `tbody > :nth-child(${nextDay[0]}) > :nth-child(${nextDay[1]})`
-    );
-  }
-}
+  const xIndex = Math.ceil(todaysIndex / 7);
+  const yIndex = day === 0 ? 7 : day;
 
-function lastDay() {
-  var currentDate = new Date(Date.now());
-
-  switch (currentDate.getMonth()) {
-    case 1:
-      return currentDate.getDate() === 29 || currentDate.getDate() === 28;
-      break;
-    case 3:
-    case 5:
-    case 8:
-    case 10:
-      return currentDate.getDate() === 30;
-      break;
-    default:
-      return currentDate.getDate() === 31;
-  }
+  return cy.get(`tbody > :nth-child(${xIndex}) > :nth-child(${yIndex})`);
```

Comment:
Is this a functionally equivalent way of getting the last day of the month including February in a leap year?
(It looks good to me, I just want to confirm)

Reply:
The remaining helper function is only for getting the current date's position.

[This commit](https://github.com/dwyl/elm-datepicker/pull/39/commits/338c9d789e83b560a51e0758906f1a9727e958e2) made an attempt to refactor and fix, but then I realised that most of the complexity was unnecessary and removed it (in the last commit). Saves the headache of trying to figure out the code, which had no functional benefit for the current set of tests.

Reply:
The combined diff is easier to understand in Split mode rather than Unified (from the gear icon on top of the diff view).